chrome/content/zotero/xpcom/openurl.js

```diff
@@ -203,25 +203,34 @@ Zotero.OpenURL = new function() {
 			}
 		}
 		
-		if(item.creators && item.creators.length) {
-			// encode first author as first and last
-			var firstCreator = item.creators[0];
-			if(item.itemType == "patent") {
-				_mapTag(firstCreator.firstName, "invfirst");
-				_mapTag(firstCreator.lastName, "invlast");
-			} else {
-				if(firstCreator.isInstitution) {
-					_mapTag(firstCreator.lastName, "aucorp");
-				} else {
-					_mapTag(firstCreator.firstName, "aufirst");
-					_mapTag(firstCreator.lastName, "aulast");
+		var first = true;
+		for(var i=0; i<item.creators.length; i++) {
+			var creator = item.creators[i];
+			// only export authors (more general primaryCreators)
+			// other creator types cannot correctly export to COinS
+			var itemTypeID = Zotero.ItemTypes.getID(item.itemType);
+			var primaryCreator = Zotero.CreatorTypes.getName(Zotero.CreatorTypes.getPrimaryIDForType(itemTypeID));
```

Comment:
I don't think that's a valid property for a translator item.
On Aug 3, 2015 12:50 AM, "Philipp Zumstein" <notifications@github.com>
wrote:

> In chrome/content/zotero/xpcom/openurl.js
> <https://github.com/aurimasv/zotero/pull/27#discussion_r36057633>:
>
> > -			if(item.itemType == "patent") {
> > -				_mapTag(firstCreator.firstName, "invfirst");
> > -				_mapTag(firstCreator.lastName, "invlast");
> > -			} else {
> > -				if(firstCreator.isInstitution) {
> > -					_mapTag(firstCreator.lastName, "aucorp");
> > -				} else {
> > -					_mapTag(firstCreator.firstName, "aufirst");
> > -					_mapTag(firstCreator.lastName, "aulast");
> > +		var first = true;
> > +		for(var i=0; i<item.creators.length; i++) {
> > +			var creator = item.creators[i];
> > +			// only export authors (more general primaryCreators)
> > +			// other creator types cannot correctly export to COinS
> > +			var itemTypeID = Zotero.ItemTypes.getID(item.itemType);
> > +			var primaryCreator = Zotero.CreatorTypes.getName(Zotero.CreatorTypes.getPrimaryIDForType(itemTypeID));
>
> Right. I will take it outside the loop.
>
> Can (and should) I write the first line as
>
> var itemTypeID = item.itemTypeID;
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aurimasv/zotero/pull/27/files#r36057633>.
>